### DISCUSSION TITLE
Helix editor

### DISCUSSION THREAD

**Discussion by stevep99** [JetBrains/ideavim#868] [General]
Are there any plans to support [Helix](https://helix-editor.com/)? Either via a selectable "hx" mode or with a forked version of the plugin with helix keybinds..

**Comment by AlexPl292** [comment]
Hi there, we have no plans for that.

**Comment by ristomatti** [comment]
I've been working on emulating Helix behavior based on [chtenb/helix.vim](https://github.com/chtenb/helix.vim). I've yet to make a pull request but I'm planning to eventually. If someone feels adventurous enough to give it a spin, you can find it at [ristomatti/helix.vim](https://github.com/ristomatti/helix.vim).

...just don't expect me to be able to fix any bugs. I'm not a Vim expert to any degree. I've pretty much just added IdeaVim specific changes/workarounds using the good old trial and error method.  :sweat_smile:

**Reply by ristomatti** [reply]
> Are there any plans to support [Helix](https://helix-editor.com/)? Either via a selectable "hx" mode or with a forked version of the plugin with helix keybinds..

@stevep99 Are you still interested in having Helix-like bindings in JetBrains IDE's? If  so, I could use some feedback on my PR at https://github.com/chtenb/helix.vim/pull/2. The maintainer is not an IdeaVim user so he can't test my changes.

**Reply by stevep99** [reply]
Despite asking the original question, I'm actually not very familiar with helix so I'm probably not the best person to judge. 

I'm familiar enough with vim to get by, but came across helix and thought the ideas behind it seemed great, so was intrigued to find out more. I was considering going all-in on helix but if we I were to do so, would need my main editors/IDE's to support it, hence the original question.

**Reply by TheBlackArroVV** [reply]
Hi, @ristomatti,
Thanks a lot for your amazing work! I started using helix fairly recently, but your work looks amazing, tbh!

The only small problem I noticed, that `w` will capture one symbol more, than it should. Example:
```
class Test
```
Helix would highlight `class ` and your implementation `class T`

Once again thanks for your job!

**Reply by ristomatti** [reply]
Thanks for the kind words @TheBlackArroVV! Sadly it's not perfect, just a workaround. However there's recently been others involved in the IdeaVim "port" who are more familiar with all this in the original repo https://github.com/chtenb/helix.vim.

I'd suggest creating a bug report there. Just please add prefix "IdeaVim:" to the issue title to differentiate since it's clear the developer who's done all the hard work is not an IdeaVim user themself. They just kindly accepted my PR :)

**Reply by citizenmatt** [reply]
Feel free to ping us if you can see that IdeaVim is doing something that it's not supposed to. That map for `w` is pretty full on! 😁 

https://github.com/chtenb/helix.vim/blob/44390d04b7aef2ba03a47789f8d9eac126934a1e/helix.idea.vim#L228

**Reply by sdoerner** [reply]
@TheBlackArroVV Please try https://github.com/chtenb/helix.vim/pull/4 , which I think will fix this issue.
I'm waiting for IdeaVim 2.19.0 to be released before merging it.

**Reply by TheBlackArroVV** [reply]
Hi, @sdoerner,
Thanks for your job, I added comment in PR

**Comment by sdoerner** [comment]
Relevant FR that should make the bindings above work better:
https://youtrack.jetbrains.com/issue/VIM-3731/Add-support-for-jump-to-previous-next-lowercase-mark
